include/openssl/types.h

```diff
@@ -229,6 +229,11 @@ typedef long ossl_intmax_t;
 typedef unsigned long ossl_uintmax_t;
 #endif
 
+/* Opaque handles to be used with core upcall functions from providers */
+typedef struct ossl_core_handle_st OSSL_CORE_HANDLE;
+typedef struct openssl_core_ctx_st OPENSSL_CORE_CTX;
+typedef struct ossl_core_bio_st OSSL_CORE_BIO;
+
```

Comment:
Hmmm...  I understand in a way, but at the same time, `core.h` contains all types that are specific to the libcrypto\<-\>provider interface.  I'm a bit ambivalent regarding spreading them out like this.  Everyone that have use of these types include `<openssl/core.h>` anyway...

Reply:
I've been dithering in my own mind between what is the correct place to have it - and have had it in both at various points. I could live with either. So I'll move it back again if you prefer.

Reply:
I think I do prefer that.

Reply:
Done